auth/server/src/main/java/com/seagates3/dao/ldap/PolicyLdapStore.java

```diff
@@ -243,4 +244,108 @@ void delete (Object policyObj) throws DataAccessException {
       throw new DataAccessException("Failed to delete the policy.\n" + ex);
     }
   }
+
+ public
+  List<Policy> findByIds(Map<String, Object> dataMap)
+      throws DataAccessException {
+    String accountName = dataMap.get("accountName") != null
+                             ? (String)dataMap.get("accountName")
+                             : "";
+    List<String> policyIds = dataMap.get("policyIds") != null
+                                 ? (List<String>)dataMap.get("policyIds")
+                                 : null;
+
+    String[] attrs = {
+        LDAPUtils.POLICY_ID,                 LDAPUtils.PATH,
+        LDAPUtils.POLICY_CREATE_DATE,        LDAPUtils.POLICY_UPDATE_DATE,
+        LDAPUtils.DEFAULT_VERSION_ID,        LDAPUtils.POLICY_DOC,
+        LDAPUtils.POLICY_NAME,               LDAPUtils.IS_POLICY_ATTACHABLE,
+        LDAPUtils.POLICY_ARN,                LDAPUtils.POLICY_ATTACHMENT_COUNT,
+        LDAPUtils.POLICY_PERMISSION_BOUNDARY};
+    // String ldapBase = LDAPUtils.BASE_DN;
+
+    String ldapBase = String.format(
+        "%s=%s,%s=%s,%s=%s,%s", LDAPUtils.ORGANIZATIONAL_UNIT_NAME,
+        LDAPUtils.POLICY_OU, LDAPUtils.ORGANIZATIONAL_NAME, accountName,
+        LDAPUtils.ORGANIZATIONAL_UNIT_NAME, LDAPUtils.ACCOUNT_OU,
+        LDAPUtils.BASE_DN);
+    String filter = "(" + LDAPUtils.OBJECT_CLASS + "=" +
+                    LDAPUtils.POLICY_OBJECT_CLASS + ")";
+    String policyIdFilter = "";
+    if (policyIds != null && !policyIds.isEmpty()) {
+      for (String policyId : policyIds) {
+        policyIdFilter += "(policyId=" + policyId + ")";
+      }
+      if (policyIds.size() > 1) {
```

Comment:
Isn't this check need to be inside for loop?

Reply:
Nope. This will be outside the for loop to create '|' condition for ldapsearch.

Reply:
So then why if (policyIds.size() > 1) {
check?
You mean if only 1 policyid is present then '|' this will not be required?
is the condition rit? > or >= ?

Reply:
yes the condition is right.. because policyIds.size() will not come <= 0 ever because the condition "if (policyIds != null && !policyIds.isEmpty()) {" is already present. So 
1. yes if 1 policyid is present then '|' is not required. 
2. policyIds size will not be 0 at this check
3. and for multiple policyIds '|' will be appended.